main.go

```diff
@@ -121,12 +91,15 @@ func serveRoot(w http.ResponseWriter, r *http.Request) {
 		)
 		http.Error(w, "parse error", http.StatusInternalServerError)
 	}
+	// TODO should this be structured and abstracted differently?
```

Comment:
not a great way to do this for every template, so worth considering how to abstract